ethcore/src/engines/clique/step_service.rs

```diff
@@ -0,0 +1,72 @@
+// Copyright 2015-2019 Parity Technologies (UK) Ltd.
+// This file is part of Parity Ethereum.
+
+// Parity Ethereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Ethereum.  If not, see <http://www.gnu.org/licenses/>.
+
+use engines::Engine;
+use parity_machine::Machine;
+use std::sync::Weak;
+use std::time::Duration;
+use std::thread;
+use std::sync::Arc;
+use parking_lot::RwLock;
+
+pub struct StepService {
+	shutdown: Arc<RwLock<bool>>,
+	thread: Option<thread::JoinHandle<()>>,
+}
+
+impl StepService {
+	pub fn start<M: Machine + 'static>(engine: Weak<Engine<M>>) -> Arc<Self> {
+		let shutdown = Arc::new(RwLock::new(false));
+		let shutdown1 = shutdown.clone();
+
+		let thread = thread::Builder::new()
+			.name("CliqueMiner".into())
+			.spawn(move || {
+				// startup delay.
+				thread::sleep(Duration::from_secs(5));
+
+				loop {
+					// see if we are in shutdown.
+					if *shutdown.read() == true {
+							trace!(target:"miner", "StepService: received shutdown signal!");
+							break;
+					}
+
+					trace!(target: "miner", "StepService: triggering sealing");
```

Comment:
I don't think the `target` field here should be "miner". It should be "engine" or "clique".

Reply:
this is actually similar to what miner does, so it make sense to use it here.